programs/clearing_house/src/math/lp.rs

```diff
@@ -0,0 +1,615 @@
+use crate::error::ClearingHouseResult;
+use crate::math::casting::cast_to_i128;
+use crate::math::constants::AMM_RESERVE_PRECISION;
+use crate::math::constants::AMM_RESERVE_PRECISION_I128;
+use crate::math::orders::standardize_base_asset_amount_with_remainder_i128;
+use crate::math_error;
+use crate::state::market::Market;
+use crate::state::market::AMM;
+use crate::state::user::MarketPosition;
+use solana_program::msg;
+
+#[derive(Debug)]
+pub struct LPMetrics {
+    pub base_asset_amount: i128,
+    pub quote_asset_amount: i128,
+    pub remainder_base_asset_amount: i128,
+    pub remainder_quote_asset_amount: i128,
+}
+
+pub fn compute_settle_lp_metrics(
+    amm: &AMM,
+    position: &MarketPosition,
+) -> ClearingHouseResult<LPMetrics> {
+    let (base_asset_amount, quote_asset_amount) = calculate_settled_lp_base_quote(amm, position)?;
+
+    // stepsize it
+    let (standardized_base_asset_amount, remainder_base_asset_amount) =
+        standardize_base_asset_amount_with_remainder_i128(
+            base_asset_amount,
+            amm.base_asset_amount_step_size,
+        )?;
+
+    let _min_qaa = amm.minimum_quote_asset_trade_size; // todo: uses reserve precision -- see note:
+    let min_baa = amm.base_asset_amount_step_size;
+
+    // note: since pnl may go into the qaa of a position its not really fair to ensure qaa >= min_qaa
+    let (remainder_base_asset_amount, remainder_quote_asset_amount) =
+        if standardized_base_asset_amount.unsigned_abs() >= min_baa {
+            // compute quote amount in remainder
+            let remainder_ratio = cast_to_i128(
+                remainder_base_asset_amount
+                    .unsigned_abs()
+                    .checked_mul(AMM_RESERVE_PRECISION)
+                    .ok_or_else(math_error!())?
+                    .checked_div(base_asset_amount.unsigned_abs())
+                    .ok_or_else(math_error!())?,
+            )?;
+
+            let remainder_quote_asset_amount = quote_asset_amount
+                .checked_mul(remainder_ratio)
+                .ok_or_else(math_error!())?
+                .checked_div(AMM_RESERVE_PRECISION_I128)
+                .ok_or_else(math_error!())?;
+
+            (remainder_base_asset_amount, remainder_quote_asset_amount)
+        } else {
+            (base_asset_amount, quote_asset_amount)
```

Comment:
here, for scenario where base_asset_amount = 0 and quote_asset_amount is deeply negative, user cannot get loss settled

Reply:
i already pushed a change to resolve (so @0xNineteen will need to update a few rust tests and review)

but it wouldnt settle since this if statement would set the full amount as remainder